api/src/main/java/org/openmrs/module/registrationcore/api/RegistrationCoreService.java

```diff
@@ -131,6 +143,15 @@
 	 */
 	List<String> findSimilarFamilyNames(String searchPhrase);
 
+	/**
+	 * Query to MPI server to find patient with Id "patientId".
+	 *
+	 * @param identifier person identifier of patient which should be imported
+	 * @param identifierTypeUuid person identifier type of patient which will be found
+	 * @return found patient
+	 */
+	Patient findMpiPatient(String identifier, String identifierTypeUuid);
```

Comment:
Does this really return Patient objects? I'd expect it to return query search results since what you get from the mpi isn't really patient objects, in fact some returned patients won't even be in the local db

Reply:
It does return a Patient object, this method is used the case you described where the returned patient is not in the local db and the user has selected to import him/her, within the importMpiPatient method. the savePatient is then called to persist the patient locally